app/controllers/registrations_controller.rb

```diff
@@ -9,11 +9,11 @@ def new
   def create
     @user = User.new(params.require(:user).permit(:email, :password, :password_confirmation, :name))
     if @user.save
-      session[:user] = @user.id
+      session[:user_id] = @user.id
       redirect_to root_path
     else
       render :new
     end
   end
 
-end
\ No newline at end of file
+end
```

Comment:
While this is the correct solution, the point of this exercise was to write a test that verifies the defective behavior before applying your solution.